### PR TITLE
Set up Travis CI job for building official Eclipse Hono Qpid Dispatch Router image

### PR DESCRIPTION
**Pull request by ppatierno**

### CODE REVIEW

File: dispatchrouter/docker-build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+then
+    TAG="$TRAVIS_TAG"
+fi
+
+docker build -t $REPO:$COMMIT $DIR || exit 1
+
+docker login -u $DOCKER_USER -p $DOCKER_PASS || exit 1
+docker tag $REPO:$COMMIT $REPO:$TAG || exit 1
+docker tag $REPO:$COMMIT $REPO:travis-$TRAVIS_BUILD_NUMBER || exit 1
+docker push $REPO || exit 1
```
Comment (sophokles73) [inline dispatchrouter/docker-build.sh]: EOL?

File: dispatchrouter/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM fedora:latest
+
+# installing needed libraries (i.e. for SASL, SSL/TLS, Python, ...)
+RUN dnf -y install cyrus-sasl-lib cyrus-sasl-plain libuuid openssl python cyrus-sasl-sql && dnf -y update && dnf clean all
+
+# adding binaries for the Qpid Proton and Qpid Dispatch Router
+ADD qpid-proton-binary.tar.gz /
+ADD qpid-dispatch-binary.tar.gz /
+
+CMD ["bash"]
```
Comment (sophokles73) [inline dispatchrouter/Dockerfile]: EOL?

File: dispatchrouter/qpid-dispatch-build/Makefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+all:
+	bash build_tarball
+
+clean:
+	rm -rf proton_build proton_install qpid-dispatch.tar.gz qpid-dispatch-src qpid-proton.tar.gz qpid-proton-src staging build qpid-dispatch-binary.tar.gz qpid-proton-binary.tar.gz
+
+.PHONY: all clean
```
Comment (sophokles73) [inline dispatchrouter/qpid-dispatch-build/Makefile]: EOL?

**Comment by ppatierno** [comment]
@sophokles73 
With Travis CI I think we don't have history of intermediate artifacts (i.e. our proton and router binaries). We should put them somewhere (i.e. a github repo ?) and if they exist avoiding re-build on next job.
Or we should decouple the steps : generate the qpid-dispatch-build image locally, run it to build proton and router artifacts, push them to the hono repo and then the Travis CI will find them during the job.
In this case we have a "manual" step to execute locally every time we know that proton and router source are changed. My objective was to avoid manual step locally.

In the current scenario for example I'd like to set two env variables in the travis file for setting proton and router version and changing the current hardcoded 0.15 and 0.7 version in the build script. When a new version is released for one (or both) of them, we can just updating this env and the new image will have new binaries automatically generated by Travis.

Discussion is open ... we have to choice the right but less complex solution :-)

**Comment by sophokles73** [comment]
I see. I guess we should start the way you have set it up. In the end, we are not blocking anybody's local build nor the builds on our Eclipse HIPP infrastructure. We can schedule the dispatch router build(s) on Travis to e.g. run every night (before the nightlies on our HIPP) and we do not really care whether they take 1 minute or 10 minutes to run, do we?

**Comment by ppatierno** [comment]
Yes this is the great thing about Travis ... nothing is running on our PCs :-)
Btw do you know a way to run scheduled Travis job at specific time ? I see "cron jobs" (as Beta feature) but no way to set up something like "every night at ...".

**Comment by sophokles73** [comment]
I thought you were the Travis guru ;-) Sorry, I have no idea ...

**Comment by ppatierno** [comment]
Sorry i'm not ... but this experience was "exciting" :-)
So can we merge this PR ? wdyt ?

**Comment by sophokles73** [comment]
Does it work? If yes, please squash and rebase it before you merge ...

**Comment by ppatierno** [comment]
Yes it works ! ... the tests as well :-)

**Pull request merged**